code/game/machinery/computer/orders/order_computer/order_computer.dm

```diff
@@ -84,19 +84,19 @@ GLOBAL_LIST_EMPTY(order_console_products)
 	data["total_cost"] = get_total_cost()
 	data["off_cooldown"] = COOLDOWN_FINISHED(src, order_cooldown)
 
-	if(!isliving(user))
-		return data
-	var/mob/living/living_user = user
-	var/obj/item/card/id/id_card = living_user.get_idcard(TRUE)
-	if(id_card)
-		data["points"] = retrive_points(id_card)
 	for(var/datum/orderable_item/item as anything in GLOB.order_console_products)
 		if(!(item.category_index in order_categories))
 			continue
 		data["item_amts"] += list(list(
 			"name" = item.name,
 			"amt" = grocery_list[item],
 		))
+	if(!isliving(user))
+		return data
+	var/mob/living/living_user = user
+	var/obj/item/card/id/id_card = living_user.get_idcard(TRUE)
+	if(id_card)
+		data["points"] = retrive_points(id_card)
```

Comment:
```suggestion
	if(isliving(user))
		var/mob/living/living_user = user
		var/obj/item/card/id/id_card = living_user.get_idcard(TRUE)
		if(id_card)
			data["points"] = retrive_points(id_card)
```

Reply:
early returning in ui_data is a risky thing to do and since it's already very close to the end anyway, I think it's better if we don't do it.